src/wallet/rpcwallet.cpp

```diff
@@ -3299,6 +3447,9 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     auto pa = keyIO.DecodePaymentAddress(fromaddress);
     fromTaddr = IsValidDestination(taddr);
     if (!fromTaddr) {
+        if (false /* Unified Address, see #5191 */) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid from address, cannot be a Unified Address, please use z_getbalanceforaddress instead.");
```

Comment:
```suggestion
            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Address cannot be a Unified Address, please use z_getbalanceforaddress instead.");
```

Reply:
Shouldn't the message still say that the problem is with the "from" address?

Reply:
This RPC is `z_getbalance`; there is no `from` address. I think this message was copy-pasted from an adjacent error, which is similarly incorrect.